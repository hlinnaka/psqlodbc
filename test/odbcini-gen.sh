#!/bin/sh
#
#	This isn't a test application.
#	Initial setting of odbc(inst).ini.
#
outini=odbc.ini
outinstini=odbcinst.ini

drvr=../.libs/psqlodbcw
driver=${drvr}.so
if test ! -e $driver ; then
	driver=${drvr}.dll
	if test ! -e $driver ; then
		echo Failure:driver ${drvr}.so\(.dll\) not found
		exit 2
	fi
fi

echo creating $outinstini
cat << _EOT_ > $outinstini
[psqlodbc test driver]
Description     = PostgreSQL ODBC driver (Unicode version), for regression tests
Driver          = $driver
Debug           = 0
CommLog         = 1
_EOT_

server=localhost
if test "${PGHOST}" != "" ; then
	server=${PGHOST}
fi
port=5432
if test "${PGPORT}" != "" ; then
	port=${PGPORT}
fi

echo creating $outini: $@
cat << _EOT_ > $outini
[psqlodbc_test_dsn]
Description             = psqlodbc regression test DSN
Driver          = psqlodbc test driver
Trace           = No
TraceFile               =
Database                = contrib_regression
Servername              = ${server}
Username                =
Password                =
Port            = ${port}
ReadOnly                = No
RowVersioning           = No
ShowSystemTables                = No
ShowOidColumn           = No
FakeOidIndex            = No
ConnSettings            =
_EOT_

# Add any extra options from the command line
for opt in "$@"
do
    echo "${opt}" >> $outini
done
