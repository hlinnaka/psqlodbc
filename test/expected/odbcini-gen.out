\! "./src/odbcini-gen.sh"
This isn't a test application
Initializing odbc(inst).ini
